app/models/plan.rb

```diff
@@ -641,6 +641,15 @@ def owner
     User.joins(:roles).where('roles.plan_id = ? AND roles.access IN (?)', self.id, vals).first
   end
 
+  ##
+  # returns the shared roles of a plan, excluding the creator
+  def shared
+    role_values = Role.access_values_for(:commenter)
```

Comment:
If you want to express "any role that is not creator" you could use the following statement:

Role.where(plan: self).where(Role.not_creator_condition) which would be less verbose

Reply:
Well, we also need to get a count of the users its shared with  (as per the  wireframe)

Reply:
`Role.not_creator_condition` returns a String for SQL querying. The results are the same for me when I run yours and Jose's to get the list of shares:
```ruby
Role.access_values_for(:creator)
# Array of Integers --> [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31]
Role.not_creator_condition
# String --> (roles.access not in (1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31))
```

Reply:
@bhavi, note you might need to exclude the reviewer role too, in that case the query should be: Role.where(plan: self).where(Role.not_creator_condition).where(Role.not_reviewer_condition).any?

otherwise a plan will be displayed as shared when someone request feedback on his/her plan